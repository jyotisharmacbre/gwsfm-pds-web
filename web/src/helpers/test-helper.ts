import checkPropTypes from 'check-prop-types';

export const findByTestAtrr = (component: any, attr: string) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const checkProps = (component: any, expectedProps: any) => {
  const propsErr = checkPropTypes(
    component.propTypes,
    expectedProps,
    'props',
    component.name
  );
  return propsErr;
};
