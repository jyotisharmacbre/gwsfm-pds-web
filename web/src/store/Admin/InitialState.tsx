import { IAdminState } from './Types/IAdminState';
import { IAdminDefaults } from './Types/IAdminDefault';

export const initialState: IAdminState = {
	adminDefaultValues: [],
	error: ''
};
