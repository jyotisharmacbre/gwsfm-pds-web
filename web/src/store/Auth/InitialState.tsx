import { IAuth } from "./Types/IAuth";

export const initialState: IAuth= {
    token: null,
    name: null,
    firstName: null,
    email: null
};
