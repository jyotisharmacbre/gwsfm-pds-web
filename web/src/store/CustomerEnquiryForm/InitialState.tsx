import { IProjectDetailState } from './Types/IProjectDetailState';
import EventType from '../../enums/EventType';
import Notify from '../../enums/Notify';

export const initialState: IProjectDetailState = {
    form: {
        projectId: '',
        name: '',
        contractorId: '',
        companyId: '',
        headOfProject: '',
        projectOwner: '',
        projectManager: '',
        pmHasExperience: false,
        scope: '',
        cnNumber: '',
        status: 1,
        engagementId: undefined,
        countryId: 0,
        currencyId: 0,
        probabilityOfWinning: '',
        approxValue: '',
        contractTypeId: '',
        cdmNotifiable: false,
        firstAssetWorkedOn: undefined,
        secondAssetWorkedOn: undefined,
        thirdAssetWorkedOn: undefined,
        comment: '',
        otherCompanyName: '',
        otherContractName: '',
        divisionId: '',
        otherDivision: '',
        businessUnitId: '',
        otherBusinessUnit: '',
        weightedTCV: undefined,
        soldMargin: undefined,
        otherEngagementType: '',
        otherFirstAssetWorkedOn: '',
        otherSecondAssetWorkedOn: '',
        otherThirdAssetWorkedOn: ''
    },
    enquiryOverview: {
        projectName: '',
        contractorId: '',
        headOfProject: '',
        projectManager: '',
        scope: '',
        cnNumber: '',
        otherContractName: ''
    },
    error: null,
    loading: false,
    notify: Notify.none,
    event: EventType.none,
    enquiryOverviewError: null
};