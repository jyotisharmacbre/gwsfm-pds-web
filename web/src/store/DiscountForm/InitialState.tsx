import { IDiscountState } from './Types/IDiscountState';
import Notify from '../../enums/Notify';
import EventType from '../../enums/EventType';
import IDiscountSubContractor from './Types/IDiscountSubContractor';

export const newSubContractorDiscount: IDiscountSubContractor = {
  subContractorDiscountId: '',
  projectId: '',
  supplierName: '',
  supplierState: '',
  supplierTotalDiscount: undefined,
  supplierComments: '',
}

export const initialState: IDiscountState = {
  form: {
    projectId: '',
    clientDiscount: {
      discountId: '',
      projectId: '',
      discount: undefined,
      clientState: '',
      discountType: 1,
      clientComments: ''
    },
    subContractorDiscounts: [{ ...newSubContractorDiscount }],
  },
  error: null,
  loading: false,
  notify: Notify.none,
  event: EventType.none
};