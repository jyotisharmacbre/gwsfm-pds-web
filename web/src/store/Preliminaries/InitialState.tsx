import Notify from '../../enums/Notify';
import { IPreliminaryState } from './Types/IPreliminaryState';
import EventType from '../../enums/EventType';
export const initialState: IPreliminaryState = {
    preliminaryDetails: [],
    notify: Notify.none,
    event: EventType.none
  };