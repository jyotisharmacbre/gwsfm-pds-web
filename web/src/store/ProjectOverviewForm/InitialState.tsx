import { IProjectOverviewState } from './Types/IProjectOverviewState';
import Notify from '../../enums/Notify';
import EventType from '../../enums/EventType';

export const initialState: IProjectOverviewState = {
  form: {
    projectId: '',
    projectAdditionalDetail: {
      projectAddDetailId: '',
      projectId: '',
      mainContractor: '',
      enquiryReceivedFrom: '',
      enquiryTypeId: -1,
      creditCheckResult: '',
      siteAddress: '',
      cdmNotifiable: false,
      formOfContract: '',
      retention: '',
      liquidatedDamages: '',
      insurance: '',
      workTypeId: -1,
      commenceDate: new Date().toJSON(),
      completionDate: new Date().toJSON(),
      milestones: '',
      firstValuationDate: new Date().toJSON(),
      finalAccountDate: new Date().toJSON(),
      valuationIntervals: '',
      paymentTerms: '',
      isProjectLive: false,
      comments: '',
      authorizedByHop: '',
      budget: 1,
      authorizedBy: '',
      authorizedBySecond: '',
      authorizedByThird: '',
      projectRisk1: '',
      projectRisk2: '',
      projectRisk3: '',
      projectRiskControlMeasure1: '',
      projectRiskControlMeasure2: '',
      projectRiskControlMeasure3: ''
    },
    projectApprovals: []
  },
  error: null,
  loading: false,
  notify: Notify.none,
  event: EventType.none
};