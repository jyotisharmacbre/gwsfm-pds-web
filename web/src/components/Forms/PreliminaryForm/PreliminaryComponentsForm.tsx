import React from 'react';
import { FieldArray} from 'redux-form';
import { FormattedMessage } from 'react-intl';
import PreliminaryItemsForm from './PreliminaryItemsForm';
import EventType from '../../../enums/EventType';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faAngleUp } from '@fortawesome/free-solid-svg-icons';
import PreliminaryInsurranceForm from './PreliminaryInsurranceForm';
import { CheckConstraints } from '../../../helpers/fieldValidations';

interface Props {
  submitHandler: (
        saveAll:boolean,
        event:EventType,
        prelimComponentDetails: any,
        index:number
      ) => void;
      onToggleEvent: (
        id: string
      ) => void;
      prelimData:any;
      handleSubmit:any;
      currencySymbol:string;
      
}  

const PreliminaryComponentsForm = ({ fields,submitHandler,handleSubmit,onToggleEvent,prelimData,currencySymbol}) => (

  <div>
  {fields.map((member, index) => (
    <div className="accordion" key={index}>
          <div className="card">
            <div
              className="card-header"
              data-toggle="collapse"
              
              
            >
              <a className="card-link" >{prelimData[index].componentName}</a>
             
              {!CheckConstraints(prelimData[index].componentId)
              ?<span aria-hidden="true">
              <FontAwesomeIcon data-test="collapse" className="active" icon={faAngleUp} onClick={()=>onToggleEvent(prelimData[index].componentId)}/>
              </span>:null}
            </div>
            <div
               id={"collapse_"+prelimData[index].componentId}
              className={CheckConstraints(prelimData[index].componentId)?"show expandAll":"hide expandAll"}
              data-test="toggle"
              data-parent="#accordion"
            > <form
            className="preliminary-form-1"
            onSubmit={handleSubmit}
            noValidate={true}
          >
                <div className="card-body pt-0">
            <div className="table-responsive price-sumry mt-0">
              <table className="table table-bordered">
                <thead>
                  <tr>
                    {!CheckConstraints(prelimData[index].componentId)?<th><FormattedMessage id="T_HEADING_ITEM" /></th>:null}
                    <th><FormattedMessage id="T_HEADING_NAME_OF_SUPPLIER" /></th>
                    {!CheckConstraints(prelimData[index].componentId)?<th><FormattedMessage id="T_HEADING_NO_OF_HOURS" /></th>:null}
                    {!CheckConstraints(prelimData[index].componentId)?<th><FormattedMessage id="T_HEADING_HOUR_RATE" /></th>:null}
                    <th><FormattedMessage id="T_HEADING_TOTAL_COST" /></th>
                    <th><FormattedMessage id="T_HEADING_GROSS_MARGIN" /></th>
                    <th><FormattedMessage id="T_HEADING_TOTAL_SELL" /></th>
                    <th><FormattedMessage id="T_HEADING_COMMENTS" /></th>
                  </tr>
                </thead>
                
                {!CheckConstraints(prelimData[index].componentId)?<FieldArray 
              name={`${member}.items`} 
              component={PreliminaryItemsForm}
              itemDetail={prelimData[index]}
              componentIndex={index}
              currencies={[]}
              currencyId={0}
              currencySymbol={currencySymbol}
              key={index}
            />:<FieldArray 
            name={`${member}.items`} 
            component={PreliminaryInsurranceForm}
            itemDetail={prelimData[index]}
            componentIndex={index}
            currencies={[]}
            currencyId={0}
            currencySymbol={currencySymbol}
            key={index}/>}
           
            </table>
                </div>
                </div>
                </form>
        <div className="text-right preliminary_btn" >
                  <button type="button" data-test="componentSave" className="text-center btn-sm" onClick={handleSubmit(values=>submitHandler(false,EventType.none,values,index))} >
                  <FormattedMessage id="BUTTON_SAVE" />
                  </button>
                </div>
             
            </div>
          </div>
     
           
        </div>
  
    ))}
    </div>
  );
 
export default PreliminaryComponentsForm;

