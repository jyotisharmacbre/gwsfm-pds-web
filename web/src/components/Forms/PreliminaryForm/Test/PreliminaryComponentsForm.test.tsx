import React from 'react';
import { shallow } from 'enzyme';
import PreliminaryComponentsForm from '../PreliminaryComponentsForm';
import { preliminariuserData, preliminaryComponentIdList } from './PreliminaryFormTestData';
import { findByTestAtrr } from '../../../../helpers/test-helper';
import { initialState } from '../../../../store/Preliminaries/InitialState';

describe('Preliminary Components Form test cases', () => {
  let wrapper: any;
  const props: any = {
    submitHandler: jest.fn(),
    onToggleEvent: jest.fn(),
    handleSubmit: jest.fn()
  };
  beforeEach(() => {
    initialState.preliminaryDetails = preliminariuserData;
    wrapper = shallow(
      <PreliminaryComponentsForm
        fields={initialState.preliminaryDetails}
        submitHandler={props.submitHandler}
        handleSubmit={props.handleSubmit}
        onToggleEvent={props.onToggleEvent}
        prelimData={initialState.preliminaryDetails}
        componentIdList={preliminaryComponentIdList}
        currencySymbol="$"
        isExpand="false"
      />
    );
  });

  it('Defines the component', () => {
    expect(wrapper).toBeDefined();
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should render toggle button event on click of component collapse icon', () => {
    const button = findByTestAtrr(wrapper, 'collapse').first();
    button.simulate('click');
    expect(props.onToggleEvent.mock.calls.length).toEqual(1);
  });
  it('should render save button once for each components', () => {
    const button = findByTestAtrr(wrapper, 'componentSave');
    expect(button.length).toEqual(1);
  });
  it('should collapse the item component on click of component collapse icon', () => {
    const button = findByTestAtrr(wrapper, 'collapse').first();
    const toggle = findByTestAtrr(wrapper, 'toggle');
    button.simulate('click');
    expect(props.onToggleEvent.mock.calls.length).toEqual(2);
  });
  it('should render check-mark if preliminariuserData item has id', () => {
    const collapseContainer = findByTestAtrr(wrapper, 'tickWrap').first();
    const checkBox = collapseContainer.find('.tick_wrap');
    expect(checkBox).toHaveLength(1);
  });
});
