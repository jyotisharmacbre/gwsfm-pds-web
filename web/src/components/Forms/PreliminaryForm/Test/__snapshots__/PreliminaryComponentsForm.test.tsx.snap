// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preliminary Components Form test cases should match the snapshot 1`] = `ReactWrapper {}`;
